Render Correct `helpdesk_email` (i.e. 'dmp-assistant@tech.alliancecan.ca') in Email Bodies

momo3404 opened the pull request:
Changes proposed in this PR:

[Refactor helpdesk_email](https://github.com/portagenetwork/roadmap/pull/1121/commits/419b2f443a9225b59608db37c08077929815d6d1)
- If `org&.helpdesk_email` was an empty string, then the previous logic of `org&.helpdesk_email || Rails.configuration.x.organisation.helpdesk_email`  would return an empty string (i.e.  `"" || dmp-assistant@tech.alliancecan.ca` returns `""`). `org&.helpdesk_email.presence || Rails.configuration.x.organisation.helpdesk_email` returns the intended `helpdesk_email` value.

[MailerHelper.helpdesk_email to `Rails.configuration.x.organisation.helpdesk_email` changes](https://github.com/portagenetwork/roadmap/pull/1121/commits/c92027e28051fdd5fe945b106733bdbbe49c12eb) 
- `MailerHelper.helpdesk_email` only uses `Rails.configuration.x.organisation.helpdesk_email` as a fallback, and instead prioritises `org&.helpdesk_email`, which could be any org's helpdesk_email. These changes directly use `Rails.configuration.x.organisation.helpdesk_email` (i.e. 'dmp-assistant@tech.alliancecan.ca') rather than `MailerHelper.helpdesk_email`.

- Breakdown of changes:
  - `app/mailers/user_mailer.rb`:
    - `MailerHelper.helpdesk_email` has been removed from all of the `UserMailer` methods
    - For `UserMailer.welcome_notification`, the updated `helpdesk_email` assignment has been moved to `app/views/user_mailer/welcome_notification.html.erb`
    - For `UserMailer.api_credentials`, the updated `helpdesk_email` assignment has been moved to `app/views/user_mailer/api_credentials.html.erb`
    - With the exception of the above two, all of the other `UserMailer` methods only reference `helpdesk_email` within `app/views/user_mailer/_email_signature.html.erb`. By assigning `helpdesk_email` within the `email_signature` partial, it is no longer needed in the `UserMailer` methods.
  - Equivalent changes to the `helpdesk_email` logic have been performed in `app/views/devise/mailer/invitation_instructions.html.erb` and `app/views/devise/mailer/reset_password_instructions.html.erb`

momo3404 committed:
Refactor helpdesk_email

aaronskiba commented on app/helpers/mailer_helper.rb:
```diff
@@ -9,7 +9,7 @@ def tool_name
 
   def helpdesk_email(org: nil)
-    org&.helpdesk_email || Rails.configuration.x.organisation.helpdesk_email
+    org&.helpdesk_email.presence || Rails.configuration.x.organisation.helpdesk_email
```
```
irb(main):001> '' || 'hello'
=> ""
```
Cool, this change will result in `Rails.configuration.x.organisation.helpdesk_email` being used as the fallback when intended.

I might just avoid titling changes like these as "refactors". This change correctly changes the prior behaviour, but a refactor should not change code behaviour.

momo3404 replied:
Sounds good, I'll change the title of the PR

aaronskiba commented:
It seems the helpdesk_email problems go deeper than this fix. Here's one example:

I set a user to a different org and for that org, I set `org.helpesk_email = 'abc@123.com'`

Next, as that user, I added a contributor to a plan. The email I added did not yet belong to any existing user in the db:
<img width="685" height="297" alt="Screenshot from 2025-08-07 12-24-56" src="https://github.com/user-attachments/assets/2b2a4ce2-efc7-4225-8bf4-fbcb8d051b10" />


This is the email that was sent to that user:

<img width="1619" height="502" alt="Screenshot from 2025-08-07 12-30-30" src="https://github.com/user-attachments/assets/2132ee0b-cb64-4214-99e7-d6816fb8fcb1" />

The email body includes the following (which I don't think is the desired email?):
- `Please do not reply to this email.  If you have any questions or need help, please contact us at [abc@123.com](mailto:abc@123.com?subject=Query%20or%20feedback%20related%20to%20DMP%20Assistant)`

aaronskiba committed:
`MailerHelper.helpdesk_email` to `Rails.configuration.x.organisation.helpdesk_email` changes

`MailerHelper.helpdesk_email` only uses `Rails.configuration.x.organisation.helpdesk_email` as a fallback, and instead prioritises `org&.helpdesk_email`, which could be any org. These changes directly use `Rails.configuration.x.organisation.helpdesk_email` rather than `MailerHelper.helpdesk_email`.

Breakdown of changes:

- `app/mailers/user_mailer.rb`:
  - `MailerHelper.helpdesk_email` has been removed from all of the `UserMailer` methods
    - For `UserMailer.welcome_notification`, the updated `helpdesk_email` assignment has been moved to `app/views/user_mailer/welcome_notification.html.erb`
    - For `UserMailer.api_credentials`, the updated `helpdesk_email` assignment has been moved to `app/views/user_mailer/api_credentials.html.erb`
    - With the exception of the above two, all of the other `UserMailer` methods only reference `helpdesk_email` within `app/views/user_mailer/_email_signature.html.erb`. By assigning `helpdesk_email` within the `email_signature` partial, it is no longer needed in the `UserMailer` methods.
  - Equivalent changes to the `helpdesk_email` logic have been performed in `app/views/devise/mailer/invitation_instructions.html.erb` and `app/views/devise/mailer/reset_password_instructions.html.erb`

momo3404 committed:
Add Changelog Entry
